### ISSUE TITLE
Doesn't work on iOS 9 and below

### ISSUE THREAD

**Issue by OCASM** [cloudrac3r/cadencegq#5]
Most likely due to lack of support of newer Javascript versions.

It's a shame because keeping old devices useful is a perfect use case for this project (that and 60fps video playback).

**Comment by cloudrac3r** [comment]
> Most likely due to lack of support of newer Javascript versions.

*Definitely* due to this. I'm using a lot of very new JS features.

iOS 9 is recent enough that I'd at least take a look at how much work would be involved in getting it working, but there's a rather large roadblock: although I do own an iOS 9 device, I can't actually bring up any sort of devtools on it, meaning I can't see any errors, meaning I have no clue what I need to change in order to make it work.

If there's a convenient way to test on iOS 9 I'll make a reasonable attempt at supporting it.

> It's a shame because keeping old devices useful is a perfect use case for this project

Sure, you could think that way, but unfortunately my views do not entirely align with yours. I like JavaScript. I like new features in JavaScript. And I like to use those new features as much as I can to make my life easier. I'm one guy.

You could argue that tools like Babel are designed for this exact situation: compiling new JavaScript to work in old browsers, but I don't really want to set up a whole workflow to rewrite all my code to support one random person's Opera Mini in iOS 5.

The official YouTube website still works in iOS 6 (an operating system that's now over 6 years old!), so if it's just about being able to play video on these devices, then you already have it.

In short, my goals are to make something cool, and I don't really care about old devices. I'm going to make it easier on myself by using new features, and I test in both Firefox and Chrome, which make up almost all of desktop market share. This project is used by so few people that supporting anything particularly old is largely a waste of time when the issue is not really with my habits but with theirs. (But who knows, maybe this will change, maybe this will become incredibly popular and I'll be motivated enough to get Babel set up and purchase more devices to test with.)

\---

That said, iOS 9 is new enough that if I can get a console window, I'll do my best to make it work.

**Comment by OCASM** [comment]
Perhaps using a desktop browser with a similar feature set? iOS 9 Safari uses Webkit 601.1. That would translate (according to Wikipedia) to OSX Safari 9. Maybe late versions of Safari 8.

Of course, it's not your fault Apple abandons perfectly usable technology. Any work on this issue would be greatly appreciated but it's understandable if you move on.

**Comment by cloudrac3r** [comment]
Checking the website "can I use" for arrow functions reveals that they are not supported in iOS 9 (or earlier) but are supported in iOS 10 and later. My code makes heavy use of arrow functions. While I could modify my code either manually or through an automated tool like Babel to convert the arrow functions to regular functions, I don't really want to.

You are of course free to fork the project and work on this yourself, but I regret to announce that I'm not willing to take the time to modify my workflow to build in Babel at the moment, and iOS 9 Safari will remain unsupported.

If you really need to use this website on iOS 9, you could try installing Chrome or Firefox, which might have better support.
